third_party/2and3/cryptography/hazmat/primitives/asymmetric/utils.pyi

```diff
@@ -1,6 +1,10 @@
-from typing import Tuple
+from typing import Optional, Tuple
+from cryptography.hazmat.primitives.hashes import HashAlgorithm
 
 def decode_dss_signature(signature: bytes) -> Tuple[int, int]: ...
 def encode_dss_signature(r: int, s: int) -> bytes: ...
 
-class Prehashed(object): ...
+class Prehashed(object):
+    _algorithm: HashAlgorithm
```

Comment:
Do we need these private attributes? We should perhaps mark them as `# undocumented`.

Also, should add the `digest_size` property.

Reply:
You are right since these are undocumented and rely on HashAlgorithm these types should simply be passed through, that is have the type of _digest_size inferred from [here](https://github.com/python/typeshed/blob/master/third_party/2and3/cryptography/hazmat/primitives/hashes.pyi#L6).

Regarding `digest_size`: I think we should also leave this to inference.

Proposed resolution: Drop the two private attributes here. Feel free to make the edit before merging, otherwise I can also update the PR, if you want.

Reply:
I don't know what you mean by "leaving to inference". Type checkers look at the stub to see what attributes `Prehashed` has, so if we don't put the public `digest_size` attribute here, type checkers won't know it exists.

Reply:
Ok, I was under the impression that `mypy` actually walks through code of libraries that have some type annotation, then it would have inferred the type of digest_size from the function signature. (see https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#type-inference)

Reply:
If there is a stub, it looks only at the stub.

Reply:
Ok, so I had a misconception regarding the tools. I'll add the `digest_size` and drop the private attributes.